homeassistant/components/__init__.py

```diff
@@ -143,9 +143,11 @@ def async_handle_turn_service(service):
         yield from asyncio.wait(tasks, loop=hass.loop)
 
     hass.services.async_register(
-        ha.DOMAIN, SERVICE_TURN_OFF, async_handle_turn_service)
+        ha.DOMAIN, SERVICE_TURN_OFF, async_handle_turn_service,
+        state_to_set=STATE_OFF)
     hass.services.async_register(
-        ha.DOMAIN, SERVICE_TURN_ON, async_handle_turn_service)
+        ha.DOMAIN, SERVICE_TURN_ON, async_handle_turn_service,
+        state_to_set=STATE_ON)
```

Comment:
Hmm. Can we use both? So for core Services that are included from `const.py` we have the interal dict lookup but the component/platform can set his own stuff on custom service without affected a core file?